frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -333,6 +336,58 @@ pub mod pallet {
                 T::DbWeight::get().reads(4)
             }
         }
+
+        #[cfg(feature = "try-runtime")]
+        fn pre_upgrade() -> Result<(), &'static str> {
+            log::info!(">>> Pre Upgrade");
+            let current_pre_approved_developers_count =
+                PreApprovedDevelopers::<T>::iter_keys().count() as u64;
+            log::info!(
+                "PreApprovedDevelopers: {}",
+                current_pre_approved_developers_count
+            );
+
+            assert_eq!(Version::V3_0_0, StorageVersion::<T>::get());
+
+            Ok(())
+        }
+
+        fn on_runtime_upgrade() -> Weight {
+            // lightweight deletions
+            let mut consumed_weight = T::DbWeight::get().writes(2);
+            StorageVersion::<T>::put(Version::default());
+            PreApprovalIsEnabled::<T>::kill();
+
+            let deletion_weight = T::DbWeight::get().writes(1) * 11 / 10;
+            match PreApprovedDevelopers::<T>::remove_all(None) {
+                KillStorageResult::AllRemoved(removed_entries_num) => {
+                    consumed_weight += deletion_weight * removed_entries_num as u64;
+                }
+                KillStorageResult::SomeRemaining(removed_entries_num) => {
+                    consumed_weight += deletion_weight * removed_entries_num as u64;
```

Comment:
I'd suggest doing an error trace instead - this mustn't happen.

Reply:
@Dinonard It seems that with polkadot-0.9.26, some changes are introduced.
1. `remove_all` has been deprecated, `clear` instead.
2. `clear` will return simple struct which includes the number of unique keys removed.

So, I updated corresponding part `on_runtime_upgrade` function. Please check.

Reply:
Looks good :+1: